lib/cable_ready/channels.rb

```diff
@@ -4,13 +4,16 @@ module CableReady
   # This class is a thread local singleton: CableReady::Channels.instance
   # SEE: https://github.com/socketry/thread-local/tree/master/guides/getting-started
   class Channels
+    include Compoundable
     extend Thread::Local
 
     def initialize
       @channels = {}
     end
 
-    def [](identifier)
+    def [](*keys)
+      keys.select!(&:itself)
+      identifier = keys.many? || (keys.one? && keys.first.is_a?(ActiveRecord::Base)) ? compound(keys) : keys.pop
```

Comment:
Just trying to grok this stream verifier stuff...

```ruby
identifier = keys.many? || <other logic>
```

So if there are multiple keys, `true` will be assigned to `identifier` here, right? Is that intended?

Reply:
Happy to have more 👀 !

Line 16 is a ternary operator... so, if there are many keys, it will resolve as true and return `compound(keys)`.

Make sense?

Reply:
I haven't finished my morning coffee yet, which usually means I will be _wildly wrong about everything_, but... are you sure?
```
irb(main):001:0> assigned_value = true || "stuff on the right hand side of this operator"
=> true
irb(main):002:0> assigned_value
=> true
```
The ternary operator on the right of the or operator won't be reached at all if `keys.multiple?` returns true, right?

Reply:
we might want to include this in one more pair of parantheses to be certain

Reply:
@Matt-Yorkley I think you need to take look what's all in your `<other logic>`.

If you would rewrite line 16 with an `if` it would look like this:

```ruby
if (keys.many? || (keys.one? && keys.first.is_a?(ActiveRecord::Base)))
 identifier = compound(keys) 
else
 identifier = keys.pop
end
```

```

Reply:
Ah right, I misread the way the brackets closed in second part. Ignore me and I'll get back to drinking my coffee :sweat_smile:

Reply:
If you add in an extra pair of parentheses, `standardrb` will remove them. I know because this is exactly what happened.

I started where Matt did, assuming that parens were require for the order of operations to do the right thing. Apparently Ruby in unambiguous about `||` and `?`.

Reply:
🤓